### PR TITLE
enable gc when evaluation

### PR DESCRIPTION
**Pull request by husimplicity**
Enable garbage collection when evaluation so that NotifyToWrite() can run

### CODE REVIEW

File: examples/igbh/dist_train_rgnn.py [hunk 1]
```diff
@@ -204,8 +204,8 @@ def run_training_proc(local_proc_rank, num_nodes, node_rank, num_training_procs,
       loss.backward()
       optimizer.step()
       total_loss += loss.item()
-      train_acc += sklearn.metrics.accuracy_score(y.cpu().numpy(),
-          out.argmax(1).detach().cpu().numpy())*100
+      train_acc += sklearn.metrics.accuracy_score(y.clone().cpu().numpy(),
+          out.argmax(1).detach().clone().cpu().numpy())*100
```
Comment (LiSu) [inline examples/igbh/dist_train_rgnn.py:208]: Unnecessary to clone the labels here? As `sklearn.metrics.accuracy_score()` only calculates the accuracy and will not cache/store the input parameters.

File: examples/igbh/dist_train_rgnn.py [hunk 2]
```diff
@@ -187,14 +187,14 @@ def run_training_proc(local_proc_rank, num_nodes, node_rank, num_training_procs,
...
-  for epoch in tqdm.tqdm(range(epochs)):
+  for epoch in range(epochs):
     model.train()
     total_loss = 0
     train_acc = 0
     idx = 0
     gpu_mem_alloc = 0
     epoch_start = time.time()
-    for batch in train_loader:
+    for batch in tqdm.tqdm(train_loader):
```
Comment (LiSu) [inline examples/igbh/dist_train_rgnn.py:197]: Revert this change? As we need to use `tqdm` to create a progress bar for epochs.

**Review by LiSu — CHANGES_REQUESTED** [review]

File: examples/igbh/dist_train_rgnn.py (hunk 1 above)
Comment (husimplicity) [reply examples/igbh/dist_train_rgnn.py]: done

File: examples/igbh/dist_train_rgnn.py (hunk 2 above)
Comment (husimplicity) [reply examples/igbh/dist_train_rgnn.py:197]: Epoch information is printed after validation. When an epoch costs much time, it can become hard to tell whether it gets stuck or is still running, so `tqdm` is moved down here.

**Review by LiSu — CHANGES_REQUESTED** [review]

File: examples/igbh/dist_train_rgnn.py (hunk 2 above)
Comment (husimplicity) [reply examples/igbh/dist_train_rgnn.py:197]: I've set the default `log_every` to 2 (can also be 1 if you think it's important) so that epoch info can be printed more frequently

**Review by LiSu — APPROVED** [review]
